### PR TITLE
[DEPR] xblock.fragment and direct id_generator parameters in xblock.runtime.Runtime

### PR DESCRIPTION
**Pull request by irtazaakram**
- Resolves https://github.com/openedx/public-engineering/issues/15

### CODE REVIEW

**Comment by feanil** [comment]
@irtazaakram it looks like there are conflicts and failing tests, can you put the PR in Draft until it's ready for review?  Or would you like a review with the tests as is?

**Comment by irtazaakram** [comment]
Hi @feanil, I've put this PR in Draft. It requires https://github.com/openedx/xblock-sdk/pull/323 this to be merged first.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,14 +7,28 @@ These are notable changes in XBlock.
 Unreleased
 ----------
 
+1.8.2 - Unreleased
```
Comment (feanil) [inline CHANGELOG.rst:10]: The version should also be bumped in `xblock/__init__.py`, also given the backwards incompatible change to the `__init__` method signature for the `Runtime` class, shouldn't this be a major version bump?

File: xblock/test/test_runtime.py [hunk 2]
```diff
@@ -650,28 +649,6 @@ def test_missing_definition(self):
             self.runtime.get_block(self.usage_id)
 
 
-class TestRuntimeDeprecation(WarningTestMixin, TestCase):
```
Comment (feanil) [inline xblock/test/test_runtime.py:653]: Why was this whole test class removed? It looks like the FieldData deprecation warnings should still be happening.

**Review by feanil — COMMENTED** [review]
I think a new version of this repo will need to be release before we can update xblock-sdk to be using that new version of the repo.

File: CHANGELOG.rst (hunk 1 above)
Comment (feanil) [reply CHANGELOG.rst]: I think this question is not resolved, given the removal, and signature change, this should be a major version bump.

Comment (irtazaakram) [reply CHANGELOG.rst]: I've already updated the version to `2.0.0` and also bumped in `xblock/__init__.py`.

https://github.com/openedx/XBlock/pull/680/files#diff-82a12ef53486e3e312a8133bf8a14c5d6be3983ed0d037ea623ca4582d3f04ec

https://github.com/openedx/XBlock/pull/680/files#diff-2c623f3c6a917be56c59d43279244996836262cb1e12d9d0786c9c49eef6b43c

**Review by feanil — APPROVED** [review]
Changes look good to me.  @farhan can you also do a review of this change?

**Comment by feanil** [comment]
Looks like there is still a couple of tests that need to be fixed as well.

**Comment by irtazaakram** [comment]
Hi @feanil ,

These tests are failing because of `xblock-sdk` as mentioned [here](https://github.com/openedx/XBlock/pull/680#issuecomment-1776618298)
> Hi @feanil,
 I've put this PR in Draft. It requires [openedx/xblock-sdk#323](https://github.com/openedx/xblock-sdk/pull/323) this to be merged first.

https://github.com/openedx/XBlock/pull/680#pullrequestreview-1694768336

> I think a new version of this repo will need to be release before we can update xblock-sdk to be using that new version of the repo.

We would need to release `0.7.1` version for xblock-sdk to clear these tests.

Thanks,

**Comment by feanil** [comment]
@irtazaakram really good point, the XBlock repo should not depend on the xblock-sdk repo, since that will cause a circular dependency that will make it hard to do incremental releases.  I've made https://github.com/openedx/XBlock/pull/688 to resolve the issue so that this repo does not depend on xblock-sdk.  Once that's merged, we should be able to rebase this PR and land it.

**Comment by feanil** [comment]
@iamsobanjaved is this ready for review?

File: .readthedocs.yaml [hunk 3]
```diff
@@ -5,6 +5,11 @@
 # Required: the version of this file's schema.
 version: 2
 
+build:
```
Comment (farhan) [inline .readthedocs.yaml:8]: Its already defined below the file

File: xblock/test/toy_runtime.py [hunk 4]
```diff
@@ -97,7 +97,7 @@ class ToyRuntime(Runtime):
     # pylint: disable=abstract-method
 
     def __init__(self, user_id=None):
-        super().__init__(ID_MANAGER, services={'field-data': KvsFieldData(TOYRUNTIME_KVS)})
+        super().__init__(ID_MANAGER, ID_MANAGER, services={'field-data': KvsFieldData(TOYRUNTIME_KVS)})
         self.id_generator = ID_MANAGER
```
Comment (farhan) [inline xblock/test/toy_runtime.py:101]: Perhaps we should remove it

**Review by farhan — CHANGES_REQUESTED** [review]

**Comment by farhan** [comment]
@irtazaakram 2 request changes rest seems good, nice work ⭐ 

Please rebase the PR

**Review by farhan — APPROVED** [review]

**Pull request merged**